### PR TITLE
Update transform example in writing-a-plugin docs

### PR DESCRIPTION
**Pull request by GAumala**
Hello, I recently wrote a gulp plugin, and while the docs were super helpful, the first example did not use the Node.js API correctly, and was a bit confusing. I hope this can make things more clear.

### CODE REVIEW

**Comment by GAumala** [comment]
Understood. Thanks for pointing that out. Made more changes.

File: docs/writing-a-plugin/README.md [hunk 1]
```diff
@@ -22,29 +22,26 @@ Transform streams are streams that are readable and writable; they manipulate ob
...
+-   */
+-  transformStream._transform = function(file, encoding, callback) {
+-    var error = null,
+-        output = doSomethingWithTheFile(file);
+-    callback(error, output);
+-  };
 ```
 
-Many plugins use the [through2](https://github.com/rvagg/through2/) module to simplify their code:
+Alternatively you could pass your transform and flush functions to the constructor or even extend `Transform` with ES6 classes, as described by the [Node.js docs](https://nodejs.org/docs/latest/api/stream.html#stream_implementing_a_transform_stream). However, many plugins prefer to use the [through2](https://github.com/rvagg/through2/) module to simplify their code:
```
Comment (phated) [inline docs/writing-a-plugin/README.md]: Don't remove the through2 recommendation.

File: docs/writing-a-plugin/README.md [hunk 2]
```diff
@@ -22,29 +22,26 @@ Transform streams are streams that are readable and writable; they manipulate ob
...
-    callback(error, output);
-  };
-
-  return transformStream;
-};
+var Transform = require('stream').Transform;
+
+-module.exports = function() {
+-  // Monkey patch Transform or create your own subclass,
+-  // implementing `_transform()` and optionally `_flush()`
```
Comment (phated) [inline docs/writing-a-plugin/README.md]: This shouldn't have changed.  Looks like you copy-pasted poorly and included `-`s and missed the end of the function.

**Comment by phated** [comment]
Closing due to inactivity.  I'll reopen if you fix it.

**Comment by GAumala** [comment]
Sorry to keep you waiting. I made more changes, please review them.

**Comment by phated** [comment]
Thanks @GAumala. I've cherry-picked the commit into master.

**Pull request closed without merging**